docker/configs/generators/generators/dot_properties.py

```diff
@@ -0,0 +1,47 @@
+import os
+import logging
+from . import *
+from cloudformation import get_stack_outputs
+
+LOGGER_NAME = os.path.splitext(os.path.basename(__file__))[0]
+LOGGER = logging.getLogger(LOGGER_NAME)
+OUTPUT_DIR = '/configs/etc/gu'
```

Comment:
The `configs` container generates all the necessary configs (`.properties` files, `nginx.conf` etc.) and shares them via a `volume` directive. I'm writing all configs to `/configs/<destination-path>` for the simplicity of sharing a single volume (`/configs`). The `entrypoint` of a container then copies the config files it needs from `/configs`.

Why is sharing one generic volume simpler than sharing many specific ones? Here's a (shortened) sequence of events I actually went through:

The `configs` container generates:
  - `nginx.conf` to `/etc/nginx/nginx.conf`
  - nginx proxy server configs to `sites-enabled/<grid-service>.conf`
  - `.properties` files to `/etc/gu`

In order to share these generated files with other containers, the directive would look like:

```yaml
configs:
  image: python:2.7
  volumes:
    - "./entrypoints:/entrypoints"
    - "./configs/generators:/configs/generators"
    - "$HOME/.aws/:/root/.aws:ro"
    - "$HOME/.gu/grid/:/root/.gu/grid"
    - "/etc/gu"
    - "/etc/nginx/nginx.conf"
    - "/etc/nginx/sites-enabled"
  entrypoint: /entrypoints/configs.sh
```

But, sharing a file (`/etc/nginx/nginx.conf`) as a volume is a bit silly! So we'd have to share `/etc/nginx/` as a whole. This means the contents of `/etc/nginx` on the `configs` container has to have all the files needed for nginx to run. Which means running (or at least installing) nginx on that container :mask: !

I then added the `imgops` container to the mix, which requires its own `nginx.conf` too, but where to put it? `/etc/nginx/imgops/nginx.conf`? :mask: !!

So I settled on creating:
  - `/configs/etc/gu/*.properties`
  - `/configs/nginx/nginx.conf`
  - `/configs/nginx/sites-enabled/*.conf`
  - `/configs/imgops/nginx.conf`

And sharing `/configs`.

Couldn't each container just generate its own config files? I started off doing this and it felt really ugly and hard to understand (and maintain) and goes against the idea of a container doing only one thing.

(Hmm... maybe I should write a blog post...)